src/routes/solid-router/reference/components/memory-router.mdx

```diff
@@ -0,0 +1,111 @@
+---
+title: MemoryRouter
+---
+
+The `MemoryRouter` can be used to route while keeping the entire routing history internally in memory.
+Other routers keep the history externally i.e. in the url like the `<HashRouter>` or in the browser history like the [`<Router>`](/solid-router/reference/components/memory-router).
+
+Keeping the history in memory is desirable when you want full control over the routing history.
+The MemoryRouter is therefore often used for testing purposes, where it's controlled through manipulating the `MemoryHistory`.
```

Comment:
```suggestion
Keeping the history in memory is useful when you want full control over the routing history.
Since `MemoryRouter` can manipulate `MemoryHistory`, it is often used for testing purposes.
```